### PR TITLE
feat: add custom actions for Workload entities

### PR DESCRIPTION
**Pull request by obs-gh-enricogiorio**
Consolidate actions into fewer files: one file per entity.

Get rid of the common interface for actions as it serves no real purpose.

Type-cast objects only once after unmarshalling (rather than once per action).

Consolidate actions' interfaces to accept the right type for such action:

 - PodAction accepts v1.Pod
 - NodeAction accepts v1.Node,
 - ... actions for new entity types will do the same

### Description

OB-XXX Please explain the changes you made here.

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: components/processors/observek8sattributesprocessor/nodeactions_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package observek8sattributesprocessor
+
+import "testing"
+
+func TestNodeActions(t *testing.T) {
+	for _, testCase := range []k8sEventProcessorTest{
+		{
+			name: "Node Status and Role",
```
Comment (obs-gh-alexlew) [inline components/processors/observek8sattributesprocessor/nodeactions_test.go:8]: kind of interesting that the test cases span across the actual actions, wouldn't it make more sense for them to be distinct? so each test case only applies to one action

Comment (obs-gh-alexlew) [reply components/processors/observek8sattributesprocessor/nodeactions_test.go:8]: is it just to reduce duplication of test events?

File: components/processors/observek8sattributesprocessor/podactions.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	PodTotalContainersAttributeKey   = "total_containers"
+	PodReadyContainersAttributeKey   = "ready_containers"
+
+	PodReadinessGatesReadyAttributeKey = "readinessGatesReady"
+	PodReadinessGatesTotalAttributeKey = "readinessGatesTotal"
+
+	PodConditionsAttributeKey = "conditions"
+)
+
+type PodStatusAction struct{}
```
Comment (obs-gh-alexlew) [inline components/processors/observek8sattributesprocessor/podactions.go:24]: nit: seems like this is missing the same header comment that other actions have

File: components/processors/observek8sattributesprocessor/types.go [hunk 3]
```diff
@@ -23,22 +27,47 @@ type K8sEventProcessorAction interface {
...
+	for _, action := range m.podActions {
+		atts, err := action.ComputeAttributes(pod)
+		if err != nil {
+			return res, err
+		}
+		res.addAttributes(atts)
 	}
-	return node, nil
+	return res, nil
 }
```
[the remark this answers is not in the document]
Comment (obs-gh-alexlew) [reply components/processors/observek8sattributesprocessor/types.go:65]: i like this better!

File: components/processors/observek8sattributesprocessor/nodeactions_test.go (hunk 1 above)
Comment (obs-gh-enricogiorio) [reply components/processors/observek8sattributesprocessor/nodeactions_test.go:8]: Mmm yeah I didn't really create these, they were already present in processor_test from previous PRs.
Maybe it's better to divide them in another commit, it's not really the purpose of this PR. I just moved them around here.

File: components/processors/observek8sattributesprocessor/podactions.go [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+		}
+	}
+
+	return attributes{
+		PodReadinessGatesTotalAttributeKey: len(pod.Spec.ReadinessGates),
+		PodReadinessGatesReadyAttributeKey: readinessGatesReady,
+	}, nil
+}
+
+// ---------------------------------- Pod "readiness" ----------------------------------
```
Comment (obs-gh-alexlew) [inline components/processors/observek8sattributesprocessor/podactions.go:171]: nit, this should be Pod "conditions" it looks like its a dupe of the header above

**Review by obs-gh-alexlew — APPROVED** [review]

**Pull request merged**